tests/tensor_search/integ_tests/test_embed.py

```diff
@@ -218,6 +265,53 @@ def run():
                 # Assert vectors are equal
                 self.assertEqual(embed_res["content"], ["I am the GOAT."])
                 self.assertTrue(np.allclose(embed_res["embeddings"][0], search_query_embedding))
+    
+    def test_embed_equivalent_to_search_text_with_prefix(self):
+        """
+            Ensure that the embedding returned by embed endpoint matches the one created by search.
+            Embedding from search is retrieved using mock on query.
+        """
+        for index in [self.unstructured_default_text_index, self.structured_default_text_index]:
+            with self.subTest(index=index.type):
+                original_query = self.config.vespa_client.query
+                def pass_through_query(*arg, **kwargs):
+                    return original_query(*arg, **kwargs)
+
+                mock_vespa_client_query = unittest.mock.MagicMock()
+                mock_vespa_client_query.side_effect = pass_through_query
+
+                @unittest.mock.patch("marqo.vespa.vespa_client.VespaClient.query", mock_vespa_client_query)
+                def run():
+                    tensor_search.search(
+                        config=self.config, index_name=index.name, text="I am the GOAT.",
+                        search_method=enums.SearchMethod.TENSOR, text_query_prefix="PREFIX: "
+                    )
+                    return True
+                self.assertTrue(run())
+
+                call_args = mock_vespa_client_query.call_args_list
+                self.assertEqual(len(call_args), 1)
+
+                vespa_query_kwargs = call_args[0].kwargs
+                if isinstance(index, UnstructuredMarqoIndex):
+                    embedding_key = "embedding_query"
+                elif isinstance(index, StructuredMarqoIndex):
+                    embedding_key = "marqo__query_embedding"
+                search_query_embedding = vespa_query_kwargs["query_features"][embedding_key]
+
+                # Embed request the same text
+                embed_res = embed(
+                    marqo_config=self.config, index_name=index.name,
+                    embedding_request=EmbedRequest(
+                        content=["I am the GOAT."],
+                        content_type="PREFIX: "
```

Comment:
Ideally, to get the same behavior, use the test model that has a prefix, and check that both embed and search use the default query prefix set for this model